src/command/verify/verify.c

```diff
@@ -1446,35 +1578,56 @@ verifyRender(const List *const archiveIdResultList, const List *const backupResu
     String *const result = strNew();
 
     // Render archive results
-    if (verboseText && lstEmpty(archiveIdResultList))
+    VariantList *archivesList = NULL;
+    VariantList *backupsList = NULL;
+    if (jsonKv != NULL)
+    {
+        archivesList = varLstNew();
+        backupsList = varLstNew();
+    }
+
+    if (jsonKv == NULL && verboseText && lstEmpty(archiveIdResultList))
+    {
         strCatZ(result, "\n  archiveId: none found");
+    }
     else
     {
         for (unsigned int archiveIdx = 0; archiveIdx < lstSize(archiveIdResultList); archiveIdx++)
         {
             const VerifyArchiveResult *const archiveIdResult = lstGet(archiveIdResultList, archiveIdx);
+            KeyValue *const archiveKv = ((archivesList == NULL) ? NULL : kvNew());
 
-            if (verboseText || archiveIdResult->totalWalFile - archiveIdResult->totalValidWal != 0)
+            if (verboseText || jsonKv != NULL || archiveIdResult->totalWalFile - archiveIdResult->totalValidWal != 0)
             {
-                strCatFmt(
-                    result, "\n  archiveId: %s, total WAL checked: %u, total valid WAL: %u", strZ(archiveIdResult->archiveId),
-                    archiveIdResult->totalWalFile, archiveIdResult->totalValidWal);
+                if (archiveKv != NULL)
+                {
+                    kvPut(archiveKv, ARCHIVE_KEY_ARCHIVEID_VAR, VARSTR(archiveIdResult->archiveId));
+                    kvPut(archiveKv, ARCHIVE_KEY_CHECKED_VAR, VARUINT(archiveIdResult->totalWalFile));
+                    kvPut(archiveKv, KEY_VALID_VAR, VARUINT(archiveIdResult->totalValidWal));
+                }
+                else
+                {
+                    strCatFmt(
+                        result, "\n  archiveId: %s, total WAL checked: %u, total valid WAL: %u", strZ(archiveIdResult->archiveId),
+                        archiveIdResult->totalWalFile, archiveIdResult->totalValidWal);
+                }
```

Comment:
It appears that the surrounding code doesn't use braces for 1-statement code blocks
```suggestion
                else
                    strCatFmt(
                        result, "\n  archiveId: %s, total WAL checked: %u, total valid WAL: %u", strZ(archiveIdResult->archiveId),
                        archiveIdResult->totalWalFile, archiveIdResult->totalValidWal);
```